src/ProtoNN/ProtoNNPredictor.cpp

```diff
@@ -9,6 +9,108 @@
 using namespace EdgeML;
 using namespace EdgeML::ProtoNN;
 
+
+ProtoNNPredictor::ResultStruct::ResultStruct()
+  :
+  problemType(undefinedProblem),
+  accuracy(0),
+  precision1(0),
+  precision3(0),
+  precision5(0)
+{}
+
+
+inline void ProtoNNPredictor::ResultStruct::scaleAndAdd(ProtoNNPredictor::ResultStruct& a, FP_TYPE scale)
+{
+  if ((problemType == undefinedProblem) && (a.problemType != undefinedProblem))
+    problemType = a.problemType;
+  assert(problemType == a.problemType);
+
+  accuracy += scale * a.accuracy;
+  precision1 += scale * a.precision1;
+  precision3 += scale * a.precision3;
+  precision5 += scale * a.precision5;
+}
+
+inline void ProtoNNPredictor::ResultStruct::scale(FP_TYPE scale)
+{
+  accuracy *= scale;
+  precision1 *= scale;
+  precision3 *= scale;
+  precision5 *= scale;
+}
+
+ProtoNNPredictor::ProtoNNPredictor(
+  const DataIngestType& dataIngestType,
+  const int& argc,
+  const char ** argv)
+{
+  assert(dataIngestType == FileIngest);
```

Comment:
@t-chgupt  This is strange. If we require fileIngest, why ask for dataIngestType. Let's just document this constructor as being useful only for when model is read from file.